enterprise/views.py

```diff
@@ -0,0 +1,184 @@
+"""
+User-facing views for the Enterprise app.
+"""
+from __future__ import absolute_import, unicode_literals
+
+import os
+
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from django.http import Http404
+from django.shortcuts import redirect
+from django.utils.translation import ugettext as _
+from django.views.generic import View
+
+try:
+    from edxmako.shortcuts import render_to_response
+except ImportError:
+    render_to_response = None
+
+try:
+    from edxmako.paths import add_lookup
+except ImportError:
+    add_lookup = None
+
+try:
+    from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+except ImportError:
+    configuration_helpers = None
+
+try:
+    from third_party_auth.pipeline import get_complete_url, get_real_social_auth_object
+except ImportError:
```

Comment:
This package has extremely strict coverage requirements; putting them all in one try/except will fail those coverage requirements during testing, because it'll jump to the except clause after the first failure. We can get around it by patching the actual imports one at a time and only leaving the last one to fail, but that's not "real" coverage.